DEVELOPING.md

```diff
@@ -1,97 +1,130 @@
-# Developing React Native
+# Developing
 
-The [example app](examples/react-native-test-suite/) allows you to test out the React Native Embrace SDK in a representative app.
+## Setup
 
-Code changes will **NOT** automatically be picked up by the example app. You **MUST** follow the instructions to test out local changes.
+```bash
+corepack enable
+yarn install
+```
 
-To get started, install node_modules with `yarn`.
+## Adding a new package
 
-React Native behaves substantially different between debug/release mode - we should always test release candidates against a release build.
+Any new package under ./packages/ will get automatically picked up as a new Yarn workspace. The directory should include:
+- `package.json` with a name and description (other fields will be filled by yarn constraints, version is supplied by Lerna)
+- `tsconfig.json` that extends from the one at the root
+- `README.md`
+- `src/` and `__tests__` folders to contain the code for the package
 
-## Run on Android
+## Adding dependencies
 
-To run on Android, follow the instructions below for debug/release mode.
+Since we are using Yarn workspaces `dependencies` should not be added to the root `package.json` (see [more details](https://stackoverflow.com/a/53558779)).
+If multiple packages include the same dependency Yarn constraints will enforce that the version used matches.
+`devDependencies` are fine to add at the root `package.json` (see [more details](https://github.com/lerna/lerna/issues/1079#issuecomment-337660289))
+if they are shared among multiple packages, otherwise add them to just the individual package that requires it.
 
-### Debug mode
+## Testing changes during development
 
-`yarn android`
+From the root of the project you can lint and test all packages with:
 
-### Release mode
+```bash
+yarn lint
+yarn test
+```
 
-`yarn android --variant=release`
+## Manual integration testing
```

Comment:
@TheMutsi I copied most of this manual testing section from the existing doc, were those steps still valid?

Reply:
I added some comments